build/pause/Makefile

```diff
@@ -14,18 +14,45 @@
 
 .PHONY:	build push
 
-TAG = 2.0
+TAG=2.0
+REGISTRY?="gcr.io/google_containers"
 
-ARCH = amd64
-IMAGE = pause
-# ARCH = arm
-# IMAGE = pause-arm
+# Architectures supported: amd64, arm, arm64 and ppc64le
+ARCH?=amd64
+GOLANG_VERSION=1.6
+TEMP_DIR:=$(shell mktemp -d)
+
+all: push
 
 build:
-	./prepare.sh ${ARCH}
-	docker build -t gcr.io/google_containers/$(IMAGE):$(TAG) .
+	cp pause.go Dockerfile $(TEMP_DIR)
+
+	docker run -it -v $(TEMP_DIR):/build golang:$(GOLANG_VERSION) /bin/bash -c \
+		"cd /build && CGO_ENABLED=0 GOARM=6 GOARCH=$(ARCH) go build -a -installsuffix cgo -ldflags '-w' ./pause.go"
+
+ifeq ($(ARCH),amd64)
+# Run goupx for amd64 and arm
+	docker run -it -v $(TEMP_DIR):/build golang:$(GOLANG_VERSION) /bin/bash -c \
+		"cd /build && apt-get update && apt-get install -y upx \
+		&& go get github.com/pwaller/goupx && goupx pause"
+endif
+ifeq ($(ARCH),arm)
```

Comment:
I hate this duplication. @zmerlynn Do you know an `if [[ cond1 || cond2 ]]; then` clause in `make`?